app/Actions/Photo/Strategies/AddBaseStrategy.php

```diff
@@ -117,45 +123,56 @@ protected function setParentAndOwnership()
 	/**
 	 * Moves/copies/symlinks source file to final destination.
 	 *
-	 * @param string $targetFullPath the path of the final destination
+	 * @param string $targetPath the path of the final destination relative to
+	 *                           the disk {@link AddBaseStrategy::IMAGE_DISK_NAME}
 	 *
 	 * @throws JsonError
 	 */
-	protected function putSourceIntoFinalDestination(string $targetFullPath): void
+	protected function putSourceIntoFinalDestination(string $targetPath): void
 	{
-		$tmpFullPath = $this->parameters->sourceFileInfo->getTmpFullPath();
-		if ($this->parameters->importMode->shallDeleteImported()) {
-			// Note: We cannot use the storage facade here, because the
-			// temporary name of the original file may be outside of the
-			// disk of the storage
-			if (!rename($tmpFullPath, $targetFullPath)) {
-				$msg = 'Could not move photo from "' . $tmpFullPath . '" to "' . $targetFullPath . '"';
-				Logs::error(__METHOD__, __LINE__, $msg);
-				throw new JsonError('$msg');
-			}
-			if (!chmod($targetFullPath, 0666 & ~umask(null))) {
-				$msg = 'Could not set permissions of "' . $targetFullPath . '"';
-				Logs::error(__METHOD__, __LINE__, $msg);
-				throw new JsonError('$msg');
-			}
-		} else {
-			// Check if the user wants to create symlinks instead of copying the photo
+		try {
+			$sourceFile = $this->parameters->sourceFileInfo->getFile();
+			$targetFile = new FlysystemFile(Storage::disk(self::IMAGE_DISK_NAME), $targetPath);
```

Comment:
Isn't this part using the `local` `s3` etc... that are defined in `config/storage.php`  ? :thinking:

Reply:
I don't understand the question.

Reply:
In `config/storage.php` you complain about the information in there such as `s3` `public` etc and are considering removing them.

But isn't `Storage::disk(DISKNAME)` making use of those information ?

Reply:
> In `config/filesystems.php` you complain about the information in there such as s3 public etc and are considering removing them.

Yes, because the are just example like a "Hello World" which are more confusing than helpful. Obviously.

> But isn't Storage::disk(DISKNAME) making use of those information?

No or at least not in the sense as you might think. `DISKNAME` **always** equals `images` and uses the `images` disk. This is a constant. You don't want to ship/compile different version of Lychee, if someone wants to switch to S3. What needs to be changed, if someone uses S3 is to configure the disk `images` appropriately.

Given this discussion, I should really remove those dummy entries from `config/filesystems.php` and instead add a proper and helpful example who to active S3 after we fully support it.

Reply:
I have cleaned up `filesystem.php` now to avoid wrong assumptions.